### PR TITLE
[Improve][Connector-V2][Socket] Unified exception for socket source & sink connector

### LINKED ISSUE #3507: [Improve][Connector-V2][Socket] Unified exception for socket source & sink connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Improve][Connector-V2][Socket] Unified exception for socket source & sink connector

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of this pull request

sub-task of #2930

close #3507

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java [hunk 1]
```diff
@@ -61,7 +62,9 @@ public void open() throws IOException {
                 createConnection();
             }
         } catch (IOException e) {
-            throw new IOException("Cannot connect to socket server at " + hostName + ":" + port, e);
+            throw new IOException(String.format(
+                "Cannot connect to socket server at %s:%d",
+                hostName, port), e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java]:
```suggestion
            throw new SocketConnectorException(SocketErrorCode.SOCKET_SERVER_CONNECT_FAILED, String.format(
                "Cannot connect to socket server at %s:%d",
                hostName, port), e);
```

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java [hunk 2]
```diff
@@ -70,37 +73,22 @@ public void write(SeaTunnelRow row) throws IOException {
...
-                throw new IOException(
-                        "Failed to send message '"
-                                + row
-                                + "' to socket server at "
-                                + hostName
-                                + ":"
-                                + port
-                                + ". Connection re-tries are not enabled.",
-                        e);
+                throw new IOException(String.format(
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java]: The same as above, you can define a new error code.

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java [hunk 3]
```diff
@@ -128,32 +116,21 @@ public void write(SeaTunnelRow row) throws IOException {
...
                     }
                     try {
                         this.wait(CONNECTION_RETRY_DELAY);
-                    }
-                    catch (InterruptedException ex) {
+                    } catch (InterruptedException ex) {
                         Thread.currentThread().interrupt();
-                        throw new IOException(
-                                "unable to write; interrupted while doing another attempt", e);
+                        throw new IOException("unable to write; interrupted while doing another attempt", e);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java]: The same as above.

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java [hunk 4]
```diff
@@ -128,32 +116,21 @@ public void write(SeaTunnelRow row) throws IOException {
...
-                                    + row
-                                    + "' to socket server at "
-                                    + hostName
-                                    + ":"
-                                    + port
-                                    + ". Failed after "
-                                    + retries
-                                    + " retries.",
-                            lastException);
+                    throw new IOException(String.format(
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java]: The same as above.

File: seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java (hunk 1 above)
Comment (wuchunfu) [reply seatunnel-connectors-v2/connector-socket/src/main/java/org/apache/seatunnel/connectors/seatunnel/socket/sink/SocketClient.java]: Done, Thanks for your review.

**Comment by wuchunfu** [comment]
@CalvinKirs @TyrantLucifer PTAL, thanks.

**Comment by TyrantLucifer** [comment]
Please update error codes in this document: https://seatunnel.apache.org/docs/connector-v2/Error-Quick-Reference-Manual

**Comment by wuchunfu** [comment]
> Please update error codes in this document: https://seatunnel.apache.org/docs/connector-v2/Error-Quick-Reference-Manual

OK

**Review by TyrantLucifer — APPROVED** [review]
+1

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**